src/browser/app/components/Input.js

```diff
@@ -0,0 +1,36 @@
+/* @flow */
+import type { Styled, InputTypes } from '../themes/types';
+import React from "react";
+import styled from "./styled";
+import Text, {TextProps} from "./Text";
+import Box from "./Box";
+
+export type InputProps = TextProps & {
+  name?: string,
+  label?: string,
+  labelSize?: string,
+  placeholder?: string,
+  maxLength?: number,
+  type?: InputTypes,
+  invalid?: boolean
+};
+
+const CustomInput = styled((theme, props: InputProps) => ({
```

Comment:
Note Styled<TextProps> pattern. Or maybe I am not sure why two inputs? Can you please describe the use case? Thank you.